src/Microsoft.AspNet.Mvc.DataAnnotations/IAttributeAdapter.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Mvc.ModelBinding.Validation;
+
+namespace Microsoft.AspNet.Mvc.DataAnnotations
+{
+    /// <summary>
+    /// Interface so that adapters provide their relevent values to error messages.
+    /// </summary>
+    public interface IAttributeAdapter : IClientModelValidator
+    {
+        /// <summary>
+        /// Implementors should provide the important values of their attribute to the base GetErrorMessage.
```

Comment:
Should just do a pass on your doc comments and add `<see...` or `<paramref...` where needed